Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -499,6 +499,22 @@ public void update(double[] point) {
         updateExecutor.update(point);
     }
 
+    /**
+     * Update the forest with the given point and a timestamp. The point is
+     * submitted to each sampler in the forest as if that timestamp was the correct
+     * stamp. storeSequenceIndexes must be false since the algorithm will not verify
+     * the correctness of the timestamp.
+     *
+     * @param point       The point used to update the forest.
+     * @param sequenceNum The timestamp of the corresponding point
+     */
+    public void update(double[] point, long sequenceNum) {
```

Comment:
This solution will let us support out-of-order updates, but it's not extensible to uses like passing a label in with the point. Do we want to consider a more generic update method?

Reply:
Yes - but perhaps not in this PR. Passing a label can be performed via constructing the label outside the forest and flattening everything into a vector. Also internal shingling may be a better (from broader usability point of view) first step for usability before categorical input.